tests/pybaseball/conftest.py

```diff
@@ -1,11 +1,17 @@
 import os
 import urllib.parse
-from typing import Callable, Dict, Generator, Union
+from typing import Callable, Dict, Generator, List, Union
 
 import pandas as pd
 import pytest
 import requests
 from _pytest.monkeypatch import MonkeyPatch
+from typing_extensions import Protocol
+
+_ParseDates = Union[bool, List[int], List[str], List[List], Dict]
+
+class GetDataFrameCallable(Protocol):
+    def __call__(self, filename: str, parse_dates: _ParseDates = False) -> pd.DataFrame: ...
```

Comment:
I'm kinda confused by this

Reply:
Protocols are a way to define that when we pass a function around (like we do in our tests to pass the functions in to load data), that the function sent over will take the correct types, and return the correct types.

So in this instance we're defining that the type `GetDataFrameCallable` is a function that will take the params defined in `__call__` and the return type defined in `__call__`.

Reply:
Can you take that comment and maybe note where this is used and make that a comment?

Reply:
Sure!